ApplicationService/connection_pool.py

```diff
@@ -0,0 +1,66 @@
+import os
+import psycopg2
+import threading
+from ApplicationService.identity import identity
+
+
+class connection_pool:
+    def get_connection(self, identifier: identity):
+        raise NotImplementedError()
+
+    def get_cursor(self, identifier: identity):
+        raise NotImplementedError()
+
+    def refund(self, identifier: identity):
+        raise NotImplementedError()
+
+class connection:
+    def cursor(self):
+        raise NotImplementedError
+
+
+class postgresql_connection_pool(connection_pool):
+    def __init__(self, createConnection=None, max_connections=1, condition=None):
+        self.create_connection = createConnection or psycopg2_create_connection
+        self.connections: dict[int, tuple] = dict()  # tuple[0] are connections and tuple[1] are conn's cursor
+        self.max_connections = max_connections
+        self.free_connections = []
```

Comment:
The only way to append free_connections is to get one and then refund. So, why not type then to list[tuple(connection,cursor)] and in the refund, don't ignore the cursor already created?